### PR TITLE
enable hot-words boosting for Javascript

### PR DESCRIPTION
**Pull request by imrahul361**
This PR enables hot-word boosting for Javascript Binaries. 
For the issue #3336

### CODE REVIEW

**Comment by ftyers** [comment]
Can you push another commit where you don't make all those spacing changes? :)

**Comment by imrahul361** [comment]
@ftyers ok sure

**Comment by imrahul361** [comment]
@ftyers it's done please check

File: native_client/javascript/client.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ parser.addArgument(['--audio'], {required: true, help: 'Path to the audio file t
 parser.addArgument(['--version'], {action: VersionAction, nargs: 0, help: 'Print version and exits'});
 parser.addArgument(['--extended'], {action: 'storeTrue', help: 'Output string from extended metadata'});
 parser.addArgument(['--stream'], {action: 'storeTrue', help: 'Use streaming code path (for tests)'});
+parser.addArgument(["--hot_words"], {help: "Hot-words and their boosts.",});
```
Comment (lissyx) [inline native_client/javascript/client.ts:34]: nit: Explain how they should be passed (commas, etc.)

File: native_client/javascript/client.ts [hunk 2]
```diff
@@ -70,6 +71,13 @@ if (args['scorer']) {
     model.setScorerAlphaBeta(args['lm_alpha'], args['lm_beta']);
   }
 }
+if (args["hot_words"]) {
```
Comment (lissyx) [inline native_client/javascript/client.ts:74]: nit: add a new line before `if`

Comment (lissyx) [reply native_client/javascript/client.ts:74]: nit: also, use single quotes for consistency

File: native_client/javascript/client.ts [hunk 3]
```diff
@@ -70,6 +71,13 @@ if (args['scorer']) {
...
+	console.error("Adding hot-words %s", args["hot_words"]);
+	for (var word_boost of args["hot_words"].split(",")) {
```
Comment (lissyx) [inline native_client/javascript/client.ts:76]: Use `let` rather than `var`

File: native_client/javascript/client.ts [hunk 4]
```diff
@@ -70,6 +71,13 @@ if (args['scorer']) {
...
+		var word = word_boost.split(":");
```
Comment (lissyx) [inline native_client/javascript/client.ts:77]: same, `let`

**Review by lissyx — COMMENTED** [review]
Thanks @imrahul361 that's a good start. Can you also add `run_hotword_tests` calls to `taskcluster/tc-node-tests.sh` and `taskcluster/tc-node_tflite-tests.sh` to cover for CI ?

File: native_client/javascript/client.ts (hunk 3 above)
Comment (imrahul361) [reply native_client/javascript/client.ts:76]: ok @lissyx  will be done in next few minutes

File: native_client/javascript/client.ts (hunk 1 above)
Comment (imrahul361) [reply native_client/javascript/client.ts:34]: will be done

**Comment by imrahul361** [comment]
> Thanks @imrahul361 that's a good start. Can you also add `run_hotword_tests` calls to `taskcluster/tc-node-tests.sh` and `taskcluster/tc-node_tflite-tests.sh` to cover for CI ?

I need to check for that once. Last time, I had tried to run the binary on local. I messed few things up.
_I need some time to be comfortable with it_

**Comment by lissyx** [comment]
> > Thanks @imrahul361 that's a good start. Can you also add `run_hotword_tests` calls to `taskcluster/tc-node-tests.sh` and `taskcluster/tc-node_tflite-tests.sh` to cover for CI ?
> 
> I need to check for that once. Last time, I had tried to run the binary on local. I messed few things up.
> _I need some time to be comfortable with it_ @lissyx

It's not intended to be run locally

**Comment by imrahul361** [comment]
> > > Thanks @imrahul361 that's a good start. Can you also add `run_hotword_tests` calls to `taskcluster/tc-node-tests.sh` and `taskcluster/tc-node_tflite-tests.sh` to cover for CI ?
> > 
> > 
> > I need to check for that once. Last time, I had tried to run the binary on local. I messed few things up.
> > _I need some time to be comfortable with it_ @lissyx
> 
> It's not intended to be run locally

ok then please guide me how can I  do that task?

File: native_client/javascript/client.ts [hunk 5]
```diff
@@ -71,6 +72,14 @@ if (args['scorer']) {
   }
 }
 
+if (args['hot_words']) {
+	console.error('Adding hot-words %s', args['hot_words']);
+	for (let word_boost of args['hot_words'].split(',')) {
+		let word = word_boost.split(':');
+		model.addHotWord(word[0], parseFloat(word[1]));
```
Comment (lissyx) [inline native_client/javascript/client.ts:79]: You actually caught a bug here, in @JRMeyer exposure to JS :)

The `index.ts` definition for `addHotWord` (and others) calls the wrong functions.

Comment (lissyx) [reply native_client/javascript/client.ts:79]: You need to fix the following call sites:
 - https://github.com/mozilla/DeepSpeech/blob/92e6daeda4196a3f99ed8414f4155787c08bf8e0/native_client/javascript/index.ts#L194
 - https://github.com/mozilla/DeepSpeech/blob/92e6daeda4196a3f99ed8414f4155787c08bf8e0/native_client/javascript/index.ts#L208
 - https://github.com/mozilla/DeepSpeech/blob/92e6daeda4196a3f99ed8414f4155787c08bf8e0/native_client/javascript/index.ts#L220

In each case, the function's first letter is not capitalized when it should be, so you need to change:
 - `binding.addHotWord` to `binding.AddHotWord`
 - `binding.eraseHotWord` to `binding.EraseHotWord`
 - `binding.clearHotWords` to `binding.ClearHotWords`

**Comment by lissyx** [comment]
@imrahul361 Note for later: can you rebase your PR this way? First commit should be the JS binding fix, then a second commit should be all your changes to add hot words to the JS client and the run of tests on CI.

**Review by lissyx — APPROVED** [review]

**Pull request merged**